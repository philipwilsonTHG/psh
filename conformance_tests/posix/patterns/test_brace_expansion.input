# Test brace expansion functionality
# PSH supports comprehensive brace expansion (v0.21.0 list, v0.22.0 sequences)

# Test basic list expansion
echo "Testing basic list expansion:"
echo {apple,banana,cherry}
echo file{1,2,3}.txt
echo {a,b,c}test

# Test list expansion with prefix/suffix
echo "Testing prefix/suffix with lists:"
echo pre{fix,post}
echo {start,end}suffix
echo pre{mid,dle}post

# Test empty elements in lists
echo "Testing empty elements:"
echo {a,,c}
echo file{,.bak}

# Test nested brace expansion
echo "Testing nested braces:"
echo {a,b{1,2},c}
echo {x,{y,z}w}
echo pre{a,{b,c}d}post

# Test numeric sequence expansion
echo "Testing numeric sequences:"
echo {1..5}
echo {10..1}
echo number{1..3}

# Test sequence with increment
echo "Testing sequences with increment:"
echo {0..10..2}
echo {10..0..2}
echo {1..9..3}

# Test zero-padded sequences
echo "Testing zero-padded sequences:"
echo {01..05}
echo {001..010}
echo file{01..03}.txt

# Test character sequences
echo "Testing character sequences:"
echo {a..e}
echo {z..v}
echo letter{A..C}

# Test mixed sequences
echo "Testing mixed case sequences:"
echo {A..C}
echo {a..c}

# Test sequence crossing zero
echo "Testing sequences crossing zero:"
echo {-3..3}
echo {-05..05}

# Test complex combinations
echo "Testing complex combinations:"
echo {a,b}{1,2}
echo {x,y}{A..C}
echo pre{1..2}{a,b}post

# Test brace expansion with glob patterns
mkdir -p tmp/brace_test
cd tmp/brace_test
touch file1.txt file2.txt file3.txt
echo "Testing braces with globs:"
echo file{1,2}.txt
echo file{1..3}.txt

# Test brace expansion in different contexts
echo "Testing braces in for loops:"
for item in {red,green,blue}; do
    echo "Color: $item"
done

# Test brace expansion with variables (should not expand)
echo "Testing braces with variables:"
var="a,b,c"
echo {$var}
echo ${var}

# Test escaped braces (should not expand)
echo "Testing escaped braces:"
echo \{a,b,c\}
echo {a\,b,c}

# Test quoted braces (should not expand)
echo "Testing quoted braces:"
echo '{a,b,c}'
echo "{a,b,c}"

# Test memory limits with reasonable expansion
echo "Testing reasonable expansion size:"
echo {1..10} | wc -w

# Test invalid sequences (should not expand)
echo "Testing invalid sequences:"
echo {z..a}
echo {1..}
echo {..5}

# Cleanup
cd ../..
rm -rf tmp/brace_test