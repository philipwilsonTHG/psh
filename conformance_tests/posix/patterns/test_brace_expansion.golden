Testing basic list expansion:
apple banana cherry
file1.txt file2.txt file3.txt
atest btest ctest
Testing prefix/suffix with lists:
prefix prepost
startsuffix endsuffix
premidpost predlepost
Testing empty elements:
a c
file file.bak
Testing nested braces:
a b1 b2 c
x yw zw
preapost prebdpost precdpost
Testing numeric sequences:
1 2 3 4 5
10 9 8 7 6 5 4 3 2 1
number1 number2 number3
Testing sequences with increment:
0 2 4 6 8 10
10 8 6 4 2 0
1 4 7
Testing zero-padded sequences:
01 02 03 04 05
001 002 003 004 005 006 007 008 009 010
file01.txt file02.txt file03.txt
Testing character sequences:
a b c d e
z y x w v
letterA letterB letterC
Testing mixed case sequences:
A B C
a b c
Testing sequences crossing zero:
-3 -2 -1 0 1 2 3
-05 -04 -03 -02 -01 000 001 002 003 004 005
Testing complex combinations:
a1 a2 b1 b2
xA xB xC yA yB yC
pre1apost pre1bpost pre2apost pre2bpost
Testing braces with globs:
file1.txt file2.txt
file1.txt file2.txt file3.txt
Testing braces in for loops:
Color:
Testing braces with variables:
a,b,c
Testing escaped braces:
{a,b,c}
a,b c
Testing quoted braces:
{a,b,c}
{a,b,c}
Testing reasonable expansion size:
      10
Testing invalid sequences:
z y x w v u t s r q p o n m l k j i h g f e d c b a
